src/ORM/Connect/MySQLStatement.php

```diff
@@ -116,6 +124,9 @@ public function nextRecord()
         // Dereferenced row
         $row = array();
         foreach ($this->boundValues as $key => $value) {
+            if ($this->types[$key] === 246) {
```

Comment:
```suggestion
            $floatTypes = [MYSQLI_TYPE_FLOAT, MYSQLI_TYPE_DOUBLE, MYSQLI_TYPE_DECIMAL, MYSQLI_TYPE_NEWDECIMAL];
            if (in_array($this->types[$key], $floatTypes)) {
```

Reply:
Squahsed this into https://github.com/silverstripe/silverstripe-framework/pull/8590/commits/45e1fcaf309bc3d3a6ae9da9a716cf4ff935b1ad